public/app/features/dashboard-scene/scene/DashboardScene.tsx

```diff
@@ -354,7 +354,20 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
       overlay: new SaveDashboardDrawer({
         dashboardRef: this.getRef(),
         saveAsCopy,
-        onSaveSuccess,
+        onSaveSuccess: () => {
+          onSaveSuccess?.();
+
+          // After saving we are updating meta to reflect runtime state of the edit options
```

Comment:
Why do we need this? It is not trivial or easy to see, maybe adding the why to the comment will help :)

Reply:
no need for that anymore :)